HBASE-23588 : Cache index & bloom blocks on write if CacheCompactedBl…

virajjasani opened the pull request:
…ocksOnWrite is enabled

virajjasani commented:
Please review @ramkrish86 @anoopsjohn

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 25s |  master passed  |
| +1 :green_heart: |  compile  |   0m 57s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 34s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 17s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 54s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 33s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 168m 55s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 226m 44s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/948 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux ea4696e31378 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-948/out/precommit/personality/provided.sh |
| git revision | master / 17e180e4ee |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/1/testReport/ |
| Max. process+thread count | 4688 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 27s |  master passed  |
| +1 :green_heart: |  compile  |   0m 55s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 34s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 16s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 14s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 52s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 35s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 192m 14s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 32s |  The patch does not generate ASF License warnings.  |
|  |   | 250m 50s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.procedure.TestDisableTableProcedure |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/948 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 2c3a556b0e27 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-948/out/precommit/personality/provided.sh |
| git revision | master / 17e180e4ee |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/2/testReport/ |
| Max. process+thread count | 4520 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

virajjasani commented on hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java:
```diff
@@ -1120,7 +1120,15 @@ public StoreFileWriter createWriterInTmp(long maxKeyCount, Compression.Algorithm
+      writerCacheConf.setCacheDataOnWrite(shouldCacheCompactedBlocksOnWrite);
+      // if data blocks are to be cached on write during compaction, we should
+      // forcefully cache index and bloom blocks as well
+      if (shouldCacheCompactedBlocksOnWrite) {
```
Agree that would be better instead of setter methods individually. Let me update

ramkrish86 replied:
@chenxu14  - Should we check for flushes also if  this has to be done ? 
@virajjasani  - You may have to add log that this been done under hood. With INFO level.

virajjasani replied:
@ramkrish86 could you please help me abort this build: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/3/

chenxu14 replied:
> Should we check for flushes also if this has to be done ?
Ya, seems reasonable.

virajjasani replied:
For flushes, it should be part of this Jira or different?

ramkrish86 commented on hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/TestCacheOnWrite.java:
```diff
@@ -462,22 +472,34 @@ private void testCachingDataBlocksDuringCompactionInternals(boolean useTags,
+
+      assertEquals(assertErrorMessage, cacheOnCompactAndNonBucketCache, dataBlockCached);
+      if (cacheOnCompactAndNonBucketCache) {
+        assertTrue(assertErrorMessage, bloomBlockCached);
```
Hope this 'if' block is getting asserted. Did you run this test case explicitly/manullay. Seems the precommit build still has this in exclude list.

virajjasani replied:
Yes I manually ran this and it passes for this if condition

virajjasani replied:
```
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.hbase.io.hfile.TestCacheOnWrite
[INFO] Tests run: 72, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 94.853 s - in org.apache.hadoop.hbase.io.hfile.TestCacheOnWrite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 72, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (secondPartTestsExecution) @ hbase-server ---
[INFO] Tests are skipped.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:58 min
[INFO] Finished at: 2019-12-19T10:51:45+05:30
[INFO] ------------------------------------------------------------------------
```

virajjasani replied:
Also confirmed that without HStore changes on this PR, assertion inside this if condition fails

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 27s |  master passed  |
| +1 :green_heart: |  compile  |   0m 56s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 40s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 11s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m  9s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 41s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 46s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 164m 50s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 222m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/948 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux dd8ee0742f0d 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-948/out/precommit/personality/provided.sh |
| git revision | master / 17e180e4ee |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/3/testReport/ |
| Max. process+thread count | 4616 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  1s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 29s |  master passed  |
| +1 :green_heart: |  compile  |   0m 55s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 41s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 23s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 21s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  1s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 44s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 25s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 167m 38s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 224m 52s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.regionserver.TestCacheOnWriteInSchema |
|   | hadoop.hbase.client.TestFromClientSide |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/948 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 0ef62e77c5e4 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-948/out/precommit/personality/provided.sh |
| git revision | master / 270eb9886e |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/4/testReport/ |
| Max. process+thread count | 4477 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 28s |  master passed  |
| +1 :green_heart: |  compile  |   0m 56s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 36s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 17s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  4s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 43s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 45s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 24s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 164m 18s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 221m 44s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/948 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 6099bb98cdbc 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-948/out/precommit/personality/provided.sh |
| git revision | master / 270eb9886e |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/5/testReport/ |
| Max. process+thread count | 4443 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/5/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

ramkrish86 commented:
LGTM. +1

virajjasani commented:
Thanks @ramkrish86 @chenxu14 
Let me have one more QA build run

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 31s |  master passed  |
| +1 :green_heart: |  compile  |   0m 55s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 16s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 14s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 22s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m  2s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 37s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 156m 38s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  The patch does not generate ASF License warnings.  |
|  |   | 214m 31s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.quotas.TestClusterScopeQuotaThrottle |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/948 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux c7445c1de32a 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-948/out/precommit/personality/provided.sh |
| git revision | master / 0ba84d8e95 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/6/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/6/testReport/ |
| Max. process+thread count | 4386 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/6/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 18s |  master passed  |
| +1 :green_heart: |  compile  |   1m 15s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 46s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 56s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 45s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 38s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 35s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 35s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 41s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 22s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 150m 36s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 34s |  The patch does not generate ASF License warnings.  |
|  |   | 212m 28s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/948 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 0646e1f95bf5 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-948/out/precommit/personality/provided.sh |
| git revision | master / e32dbe8ed2 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/8/testReport/ |
| Max. process+thread count | 4906 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-948/8/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

anoopsjohn commented on hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java:
```diff
@@ -1120,9 +1120,25 @@ public StoreFileWriter createWriterInTmp(long maxKeyCount, Compression.Algorithm
       // Don't cache data on write on compactions, unless specifically configured to do so
       writerCacheConf = new CacheConfig(cacheConf);
-      writerCacheConf.setCacheDataOnWrite(cacheConf.shouldCacheCompactedBlocksOnWrite());
+      final boolean shouldCacheCompactedBlocksOnWrite = cacheConf
```
Sorry for being late here.
The var name here sounds like a method name. Ideally it should be a noun. 'cacheCompactedBlocksOnWrite'

anoopsjohn commented on hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java:
```diff
@@ -1120,9 +1120,25 @@ public StoreFileWriter createWriterInTmp(long maxKeyCount, Compression.Algorithm
+      // forcefully cache index and bloom blocks as well
+      if (shouldCacheCompactedBlocksOnWrite) {
+        writerCacheConf.enableCacheOnWrite();
+        LOG.info("cacheCompactedBlocksOnWrite is true, hence enabled cacheOnWrite for " +
```
IMHO this will create so much INFO log each of which wont be giving much meaningful info later. No info which file being written.  At the RS level we can log once in INFO level that because of enableCacheOnCompaction is true, we make all 3 to true. But now it will be logged for each and every flush and compaction.

virajjasani commented:
Thanks @anoopsjohn 
Since this is merged, let me add addendum patch addressing your review comments. Agree on var name change and on too many INFO logs, let me add trace log may be?
